src/components/Tables/TableFilterBar.tsx

```diff
@@ -1,33 +1,70 @@
 "use client";
 
 import React from "react";
+import { styled } from "styled-components";
+import FilterAccordion from "./Accordion";
 
 export interface FilterText {
     [key: string]: string;
 }
+const StyledFilterAccordion = styled(FilterAccordion)`
+    width: 100%;
+    overflow: visible;
+`;
+
+const ClearButton = styled.button`
+    background-color: ${(props) => props.theme.secondaryBackgroundColor};
+    box-shadow: 0 0 1px ${(props) => props.theme.secondaryBackgroundColor};
+    height: 3rem;
+    width: 6rem;
+    font-size: 1rem;
+`;
+
+const StyledFilterBar = styled.input`
+    height: 3rem;
+    font-size: 1rem;
+    width: 15rem;
+    overflow: visible;
+    border: 1px solid ${(props) => props.theme.secondaryBackgroundColor}!important;
+    box-shadow: 0 0 1px ${(props) => props.theme.secondaryBackgroundColor};
+    padding: 4px 4px 4px 8px !important;
+    border-radius: 0.8rem;
+`;
 
 interface Props {
     filterText: FilterText;
     onFilter: (event: React.ChangeEvent<HTMLInputElement>, filterField: string) => void;
     handleClear: () => void;
-    headers: FilterText;
+    headers: [string, string][];
+    filterKeys: string[];
+    toggleableHeaders?: string[];
+    setShownHeaderKeys: (headers: string[]) => void;
+    shownHeaderKeys: string[];
 }
 
 const TableFilterBar: React.FC<Props> = (props) => {
     return (
         <>
-            {Object.entries(props.headers).map(([key, value]) => {
-                return (
-                    <input
-                        key={key}
-                        type="text"
-                        value={props.filterText[key] ?? ""}
-                        placeholder={`Filter by ${value}`}
-                        onChange={(event) => props.onFilter(event, key)}
-                    />
-                );
-            })}
-            <button onClick={props.handleClear}>Clear</button>
+            <ClearButton onClick={props.handleClear}>Clear</ClearButton>
+            {props.headers
+                .filter(([key, _value]) => props.filterKeys?.includes(key) ?? true)
+                .map(([key, value]) => {
+                    return (
+                        <StyledFilterBar
```

Comment:
I tried the Filter and its currently doing nothing. Is this currently being fixed because the table filter functionality was working before.

Reply:
Fixed